### DISCUSSION TITLE
compiling with docker question

### DISCUSSION THREAD

**Discussion by gartnl** [egzumer/uv-k5-firmware-custom#258] [Q&A] [answered]
Ok, not really about the firmware but:
My old machine I usually use for arduino, jscript, webapps and purebasic did not want to install winget ( nor the app-store ).
So I wanted to give docker a try. Docker seems installed ok. Downloaded repository and unzipped. Ran compile-with-docker.bat.
Docker seemed to download all stuff it needs until I got the error: >fatal: not a git repository (or any of the parent directories): .git<

Must be something I miss as a docker newbie. Advice in the right direction is appreciated.

**Comment by egzumer** [comment]
You probably downloaded zipped project. You probably need to git clone it.

**Comment by egzumer** [comment]
There is another way of building it you might want to try. 
Go to https://github.com/egzumer/uv-k5-firmware-custom
Hit green Code button
switch a tab from local to codespace
hit create codespace on main
you will get terminal window, run in it: ./compile-with-docker.sh
on the left you will have files, search for compiled-firmware folder

**Reply by gartnl** [reply]
Tnx. I indeed downloaded the zip. Suspected that that's was the course. I'll try your second suggestion tomorrow.
(And start searching for a better machine for my local experiments)

**Comment by egzumer** [comment, marked as answer]
https://github.com/egzumer/uv-k5-firmware-custom?tab=readme-ov-file#github-codespace-build-method

**Comment by gartnl** [comment]
Well, it was quit a steep learning curve but got my own build working. 
And all of that for a minor change: wanted to get rid of the diagonal in the '0' on the frequency display. My old eyes kept misreading it for an '8' without reading glasses.
So it is now V0.20.P for my personal pride ..

**Reply by egzumer** [reply]
You have my respect for putting your own effort, congrats.

**Comment by ad6xp** [comment]
The **github codespace build method failed** for me on a first try.
The problem arised with a installation of base-devel package:
4.855  base-devel-1-1-any downloading...
4.856 checking keyring...
6.550 checking package integrity...
9.302 error: base-devel: signature from "Levente Polyak (anthraxx) <levente@leventepolyak.net>" is unknown trust
9.303 error: failed to commit transaction (invalid or corrupted package)
9.303 :: File /var/cache/pacman/pkg/base-devel-1-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
9.303 Do you want to delete it? [Y/n] 
9.345 Errors occurred, no packages were upgraded.

By adding a line `pacman-key --refresh-keys` to a Dockerfile the error resolved. 
The key refreshment took some time but next build was fast.

This is a copy of my Dockerfile:

```
FROM --platform=amd64 archlinux:latest
RUN pacman-key --refresh-keys
RUN pacman -Syyu base-devel --noconfirm
RUN pacman -Syyu arm-none-eabi-gcc --noconfirm
RUN pacman -Syyu arm-none-eabi-newlib --noconfirm
RUN pacman -Syyu git --noconfirm
RUN pacman -Syyu python-pip --noconfirm
RUN pacman -Syyu python-crcmod --noconfirm
WORKDIR /app
COPY . .
RUN git submodule update --init --recursive
```
Happy coding in 2024!
Tom

**Reply by egzumer** [reply]
This command executes horribly long so I don't know what to do with this. From what I can find on the Internet it should never be needed. This looks like some packages need to be updated in archlinux.

**Reply by egzumer** [reply]
They fixed the problem at the source (updated package signature). It works now.
